lib/ansible/modules/network/netscaler/netscaler_nitro_request.py

```diff
@@ -0,0 +1,881 @@
+    returned: always
+    type: string
+    sample: "{ errorcode: 0, message: Done, severity: NONE }"
+
+nitro_object:
+    description: The object returned from the NITRO operation. This is applicable to the various get operations which return an object.
+    returned: when applicable
+    type: list
+    sample:
+        -
+            ipaddress: "192.168.1.8"
+            ipv6address: "NO"
+            maxbandwidth: "0"
+            name: "test-server-1"
+            port: 0
+            sp: "OFF"
+            state: "ENABLED"
+
+nitro_auth_token:
+    description: The token returned by the C(mas_login) operation when succesful.
+    returned: when applicable
+    type: string
+    sample: "##E8D7D74DDBD907EE579E8BB8FF4529655F22227C1C82A34BFC93C9539D66"
+'''
+
+
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.basic import env_fallback
+from ansible.module_utils.basic import AnsibleModule
+import codecs
+
+
+class NitroAPICaller(object):
+
+    _argument_spec = dict(
+        nsip=dict(
+            fallback=(env_fallback, ['NETSCALER_NSIP']),
+        ),
+        nitro_user=dict(
+            fallback=(env_fallback, ['NETSCALER_NITRO_USER']),
+        ),
+        nitro_pass=dict(
+            fallback=(env_fallback, ['NETSCALER_NITRO_PASS']),
+            no_log=True
+        ),
+        nitro_protocol=dict(
+            choices=['http', 'https'],
+            fallback=(env_fallback, ['NETSCALER_NITRO_PROTOCOL']),
+            default='http'
+        ),
+        validate_certs=dict(
+            default=True,
+            type='bool'
+        ),
+        nitro_auth_token=dict(
+            type='str',
+            no_log=True
+        ),
+        resource=dict(type='str'),
+        name=dict(type='str'),
+        attributes=dict(type='dict'),
+
+        args=dict(type='dict'),
+        filter=dict(type='dict'),
+
+        operation=dict(
+            type='str',
+            required=True,
+            choices=[
+                'add',
+                'update',
+                'get',
+                'get_by_args',
+                'get_filtered',
+                'get_all',
+                'delete',
+                'delete_by_args',
+                'count',
+
+                'mas_login',
+
+                # Actions
+                'save_config',
+
+                # Generic action handler
+                'action',
+            ]
+        ),
+        expected_nitro_errorcode=dict(
+            type='list',
+            default=[0],
+        ),
+        action=dict(type='str'),
+        instance_ip=dict(type='str'),
+        instance_name=dict(type='str'),
+        instance_id=dict(type='str'),
+    )
+
+    def __init__(self):
+
+        self._module = AnsibleModule(
+            argument_spec=self._argument_spec,
+            supports_check_mode=False,
+        )
+
+        self._module_result = dict(
+            changed=False,
```

Comment:
This doesn't seem right

Reply:
The changed status was not properly set for all operations.

I corrected that with b493902.

Now the `changed` value is set according to operation. 

Read operations like `get` and `count` set `changed` to `False`. 

Modify operations like `add`, `update`, `delete` set a `changed` status to `True` when successful and to `False` when not successful.

@gundalow  thanks for catching that.